server/src/models/gorilla.rs

```diff
@@ -264,3 +281,258 @@ impl<'a> Bits<'a> {
+        let expected_sum = aggregate::sum(&ValueArray::from_iter_values(values)).unwrap();
+        prop_assert!(equal_or_nan(expected_sum, sum));
+    }
+    }
+
+    // Tests for grid().
+    proptest! {
+    #[test]
+    fn test_grid(values in collection::vec(ProptestValue::ANY, 0..50)) {
+        prop_assume!(!values.is_empty());
+        let compressed_values = compress_values_using_gorilla(&values);
+        let mut tids_builder = TimeSeriesIdBuilder::new(10);
+        let mut timestamps_builder = TimestampBuilder::new(10);
+        let mut values_builder = ValueBuilder::new(10);
+
+        grid(
+            1,
+            1,
+            values.len() as i64,
+            1,
+            &compressed_values,
+            &[],
+            &mut tids_builder,
+            &mut timestamps_builder,
+            &mut values_builder
+        );
+
+        let tids_array = tids_builder.finish();
+        let timestamps_array = timestamps_builder.finish();
+        let values_array = values_builder.finish();
+
+        prop_assert!(
+            tids_array.len() == values.len()
+            && tids_array.len() == timestamps_array.len()
+            && tids_array.len() == values_array.len()
+        );
+        prop_assert!(tids_array.iter().all(|tid_option| tid_option.unwrap() == 1));
+        prop_assert!(timestamps_array.values().windows(2).all(|window| window[1] - window[0] == 1));
+        prop_assert!(slice_of_value_equal(values_array.values(), &values));
+    }
+    }
+
+    // Tests for the decompress_values().
+    proptest! {
+    #[test]
+    fn test_decode(values in collection::vec(ProptestValue::ANY, 0..50)) {
+        prop_assume!(!values.is_empty());
+        let compressed_values = compress_values_using_gorilla(&values);
+        let mut decompressed_values_builder = ValueBuilder::new(values.len());
+        decompress_values(
+            1,
+            values.len() as i64,
+            1,
+            &compressed_values,
+            &mut decompressed_values_builder
+        );
+        let decompressed_values = decompressed_values_builder.finish();
+        prop_assert!(slice_of_value_equal(decompressed_values.values(), &values));
+    }
+    }
+
+    fn compress_values_using_gorilla(values: &[Value]) -> Vec<u8> {
+        let mut model_type = Gorilla::new();
+        for value in values {
+            model_type.compress_value(*value);
+        }
+        model_type.get_compressed_values()
+    }
+
+    fn slice_of_value_equal(values_one: &[Value], values_two: &[Value]) -> bool {
+        let mut equal = true;
+        for values in values_one.iter().zip(values_two) {
+            let (value_one, value_two) = values;
+            equal &= equal_or_nan(*value_one, *value_two);
+        }
+        equal
+    }
+
+    fn equal_or_nan(v1: Value, v2: Value) -> bool {
+        v1 == v2 || (v1.is_nan() && v2.is_nan())
+    }
+
+    // Tests for BitReader.
+    #[test]
+    fn test_empty_bit_reader_error() {
```

Comment:
Maybe rename to "test_bit_reader_cannot_be_empty" or something similar for consistency with other `is_err()` tests.

Reply:
Fixed.